templates/web/pages/product_edit/product_edit_form_packagings.tt.html

```diff
@@ -0,0 +1,182 @@
+<!-- start templates/[% template.name %] -->
+
+<initjs>
+var packaging_shapes_select2_options = [% generate_select2_options_for_taxonomy_to_json("packaging_shapes") %];
+var packaging_materials_select2_options = [% generate_select2_options_for_taxonomy_to_json("packaging_materials") %];
+var packaging_recycling_select2_options = [% generate_select2_options_for_taxonomy_to_json("packaging_recycling") %];
+
+var component = 0;
+</initjs>
+
+<p>[% lang('packaging_parts') %]</p>
+    
+<div id="packagings_components_edit">
+
+    <!-- header row for large display -->
+    <div class="row show-for-large-up" id="packagings_components_edit_header">
+        <div class="large-1 columns">[% lang("packaging_number_of_units") %]</div>
+        <div class="large-3 columns">[% lang("packaging_shape") %]</div>
+        <div class="large-3 columns">[% lang("packaging_material") %]</div>
+        <div class="large-3 columns">[% lang("packaging_recycling") %]</div>
+        <div class="large-1 columns">[% lang("packaging_weight") %] (g)</div>
+        <div class="large-1 columns">[% lang("packaging_quantity_per_unit") %]</div>
+    </div>
+
+[% FOREACH packaging IN product.packagings %]
+    [% SET packaging_max = loop.index() %]
```

Comment:
why not `loop.max()` ?

Reply:
ok maybe it's more clear if you write:
```suggestion
    [% SET packaging_index = loop.index() %]
```